packages/flutter/lib/src/material/input_decorator.dart

```diff
@@ -2405,8 +2405,7 @@ class _InputDecoratorState extends State<InputDecorator> with TickerProviderStat
       floatingLabelHeight = 0.0;
       contentPadding = decorationContentPadding ?? EdgeInsets.zero;
     } else if (!border.isOutline) {
-      // 4.0: the vertical gap between the inline elements and the floating label.
-      floatingLabelHeight = (4.0 + 0.75 * labelStyle.fontSize!) * MediaQuery.textScaleFactorOf(context);
+      floatingLabelHeight = ((decoration.floatingLabelGap ?? 4.0) + 0.75 * labelStyle.fontSize!) * MediaQuery.textScaleFactorOf(context);
```

Comment:
This means that your floatingLabelGap will change with the text scale factor.  So if a users sets their system font size to larger than normal in their device settings, then the gap will increase.  Do you think that's desired?

If we change it so that text scale factor doesn't affect this gap, then likely some visual diff tests will be broken.

Reply:
Also could the `4.0` be set as the default value for the floatingLabelGap parameter?

Reply:
> if a users sets their system font size to larger than normal in their device settings, then the gap will increase. Do you think that's desired?

This is what happens now. My change reflex as minimal alternation as possible.

Reply:
> Also could the `4.0` be set as the default value for the floatingLabelGap parameter?

I think it could be, but it was simpler to code this way. I don't remember the issue I ran into when I tried it that way first but I switched it to this because I hit a snag.